azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchManager.java

```diff
@@ -320,6 +323,25 @@ public void run() {
     }
   }
 
+  @Override
+  public void cancelFlow(ExecutableFlow exFlow, String userId)
+      throws ExecutorManagerException {
+    synchronized (exFlow) {
+      final Map<Integer, Pair<ExecutionReference, ExecutableFlow>> unfinishedFlows = this.executorLoader
+          .fetchUnfinishedFlows();
+      if (unfinishedFlows.containsKey(exFlow.getExecutionId())) {
+        final Pair<ExecutionReference, ExecutableFlow> pair = unfinishedFlows
+            .get(exFlow.getExecutionId());
+        // Note that ExecutionReference may have the 'executor' as null. ApiGateway call is expected
+        // to handle this scenario.
+        this.apiGateway.callWithReferenceByUser(pair.getFirst(), ConnectorParams.CANCEL_ACTION, userId);
+      } else {
+        throw new ExecutorManagerException("Execution "
```

Comment:
Can you please add error log over here? I also miss this alot :-)

Reply:
Fixed.